### DISCUSSION TITLE
Fastfetch

### DISCUSSION THREAD

**Discussion by contrarybaton60** [ivan-hc/AM#1081] [General]
I didn't knew that Fastfetch is able to detect AM packages
![Screenshot_debian12xfce_2024-11-05_08:46:56](https://github.com/user-attachments/assets/2452cf23-a6a2-42a0-9025-140c6419ddd5)

**Comment by Samueru-sama** [comment]
I requested it [here](https://github.com/fastfetch-cli/fastfetch/issues/771)

However fastfetch does not detect appman packages.

**Reply by ivan-hc** [reply]
@Samueru-sama normal behaviour in AM is this

![Istantanea_2024-12-06_04-32-32](https://github.com/user-attachments/assets/555014de-9a1b-4252-a736-2cd6f2b8b926)

it lists both system and local programs.

It would be enough to use this command to list all programs installed locally and system wide
```
_determine_args && echo "$ARGS" | uniq | wc -l
```
![Istantanea_2024-12-06_04-35-27](https://github.com/user-attachments/assets/6e846e91-6a4e-40f0-945d-a6ad2591d984)

Better if
```
_determine_args && echo "$ARGPATHS" | uniq | wc -l
```
to have the full path of the installed program, so you can add a `grep -v "^/opt/"` to list only local apps.

![Istantanea_2024-12-06_04-37-09](https://github.com/user-attachments/assets/5b741bbb-d57f-4b11-9d8d-50356c0abce5)

----------------------

That said, I don't know what is the behaviour you would see... I only know that I have 8 system programs (AM included) and `fastfetch` lists only 7 of them... but I don't know how it sees the number of programs. Not all of them have an AM-updater script, for example Firefox

**Reply by ivan-hc** [reply]
this is the function in APP-MANAGER that lists all programs
```
function _detect_appman_apps() {
	[ -f "$APPMANCONFIG/appman-config" ] && APPMAN_APPSPATH=$(<"$APPMANCONFIG/appman-config")
	if ! echo "$APPMAN_APPSPATH" | grep -q "^/"; then
		[ -f "$APPMANCONFIG/appman-config" ] && APPMAN_APPSPATH="$HOME/$(<"$APPMANCONFIG/appman-config")"
	fi
}

function _determine_args() {
	ARGPATHS=$(find "$APPSPATH" -maxdepth 2 -name 'remove' -printf " %h\n" 2>/dev/null | sort -u | sed 's/ //g')
	ARGS=$(echo "$ARGPATHS" | xargs -n 1 basename 2>/dev/null)
	if [ "$AMCLI" = am ]; then
		_detect_appman_apps
		if [ -d "$APPMAN_APPSPATH" ]; then
			APPMAN_PATHS=$(find "$APPMAN_APPSPATH" -maxdepth 2 -name 'remove' -printf " %h\n" 2>/dev/null | sort -u | sed 's/ //g')
			ARGPATHS=$(echo -e "$ARGPATHS\n$APPMAN_PATHS")
			ARGS=$(echo "$ARGPATHS" | xargs -n 1 basename 2>/dev/null)
		fi
	fi
	# use "argpath=$(echo "$ARGPATHS" | grep "/$arg$")" to determine the full path of "arg"
}
```
